### PR TITLE
Pep 777 updates

### PR DESCRIPTION
**Pull request by emmatyping-nv**

### CODE REVIEW

File: peps/pep-0777.rst [hunk 1]
```diff
@@ -7,7 +7,7 @@ Status: Draft
 Type: Standards Track
 Topic: Packaging
 Created: 09-Oct-2024
-Post-History:
+Post-History: `10-Oct-2024 <https://discuss.python.org/t/pep-777-how-to-re-invent-the-wheel/67484>`_
```
Comment (warsawnv) [inline peps/pep-0777.rst:11]: Interesting.  I didn't realize PEP 1 had been updated to include a link in the `Post-History`.  At the least you should probably also add a `Discussions-To` header.

File: peps/pep-0777.rst [hunk 2]
```diff
@@ -76,22 +76,47 @@ However, resolvers do not currently exclude wheels with an incompatible wheel
...
-The ``Wheel-Version`` field in the metadata file shall contain the exact same entry as the
-``Wheel-Version`` entry in the ``WHEEL`` file, or any future replacement file
-defining metadata about the wheel file. Installers **MUST** verify that these
-entries match when installing a wheel. If ``Wheel-Version`` is absent from the
-metadata file, then the implied major version of the wheel is 1.
+the ``Wheel-Version`` field. Build backends are encouraged to include
+``Wheel-Version`` in all wheels that they generate, regardless of version.
+
+When installing a wheel, installers **MUST NOT** remove the ``Wheel-Version``
+in the installed metadata file. If this was allowed, the ``RECORD`` file would
```
Comment (warsawnv) [inline peps/pep-0777.rst]: I don't follow the "if this was allowed" reasoning and tbh don't remember this issue from the discussion thread.  Could you elaborate or add a link to the deeper discussion about why this must be so?

File: peps/pep-0777.rst [hunk 3]
```diff
@@ -211,6 +238,14 @@ contents. Wheel revisions shouldn't cause package installations to break on
 older CPython revisions, as not only would it be frustrating, it would be
 incredibly hard to debug for users.
 
+This PEP relies on resolvers being able to efficiently acquire package
+metadata, usually through :pep:`658`. This might present a problem for users of
+package indices that do not serve :pep:`658` metadata. However, today most
```
Comment (warsawnv) [inline peps/pep-0777.rst:243]: The text is not clear why the "However" sentence is relevant to the reliance on acquiring metadata via 658 or not.  Perhaps elaborate on the connection between range requests and efficient access to the metadata?

File: peps/pep-0777.rst [hunk 4]
```diff
@@ -266,6 +301,61 @@ build tag in the middle of the file name makes any extensions ambiguous (i.e.
...
+and these would need to be updated every version that is released. Furthermore,
+part of the brittleness of the current wheel specification is that so much
+metadata is stored in the filename. Filenames are not well suited to store
+structured data. Moving away from encoding information in the filename should
+be a goal of future wheel revisions.
+
+Another possibility is to use the file extension to encode the outer container
+format (i.e. a ZIP file containing ``.dist-info``) separate from the inner
+wheel version. However, this could lead to confusion over a difference in
+`Wheel-Version` vs the version in the extension. If an installer errors that
```
Comment (warsawnv) [inline peps/pep-0777.rst]: The implication here being that the extension may not always need to track the wheel version number, if e.g. future extensions can be handled in way that's backward compatible with the outer container format.  Am I expressing that correctly?  I agree with the rejection of this point, but I feel like there's perhaps a logic step that could be strengthened.

**Review by warsawnv — APPROVED** [review]
It's a great update, thanks.  I had a few comments and questions, but no blockers.  In the future, if you'd prefer to PR against the real PEPs repo, that would be fine by me.  Totally up to you though!

**Comment by emmatyping-nv** [comment]
Thank you for the review! I'll make a PR against the PEPs repo.

**Pull request closed without merging**